src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -424,8 +424,12 @@ private static MediaTypeCollection GetDeclaredContentTypes(IApiRequestMetadataPr
                 foreach (var metadataAttribute in responseMetadataAttributes)
                 {
                     metadataAttribute.SetContentTypes(contentTypes);
-
-                    if (metadataAttribute.Type != null)
+                    if (metadataAttribute.Type == typeof(void) && type != null
```

Comment:
@rynowak should this go behind a compat flag?

Reply:
No, we don't use compat switches for patches